llvm/include/llvm/Analysis/TargetTransformInfo.h

```diff
@@ -1857,6 +1857,10 @@ class TargetTransformInfo {
   /// \returns True if the target supports scalable vectors.
   LLVM_ABI bool supportsScalableVectors() const;
 
+  /// \returns True if the target supports speculative load intrinsics (e.g.,
+  /// vp.load.ff).
+  LLVM_ABI bool supportsSpeculativeLoads() const;
```

Comment:
I think you probably need to pass a type into this interface because the target may not support speculative loads of 128-bit types, etc.

Reply:
Great point! I missed that different types may have different support levels. Thanks for the feedback. Fixed.